v1/plugins/logs/plugin.go

```diff
@@ -249,23 +249,29 @@ func roundtripJSONToAST(x interface{}) (ast.Value, error) {
 
 const (
 	// min amount of time to wait following a failure
-	minRetryDelay                       = time.Millisecond * 100
-	defaultMinDelaySeconds              = int64(300)
-	defaultMaxDelaySeconds              = int64(600)
-	defaultUploadSizeLimitBytes         = int64(32768) // 32KB limit
-	defaultBufferSizeLimitBytes         = int64(0)     // unlimited
-	defaultMaskDecisionPath             = "/system/log/mask"
-	defaultDropDecisionPath             = "/system/log/drop"
-	logRateLimitExDropCounterName       = "decision_logs_dropped_rate_limit_exceeded"
-	logNDBDropCounterName               = "decision_logs_nd_builtin_cache_dropped"
-	logBufferSizeLimitExDropCounterName = "decision_logs_dropped_buffer_size_limit_bytes_exceeded"
-	logEncodingFailureCounterName       = "decision_logs_encoding_failure"
-	defaultResourcePath                 = "/logs"
+	minRetryDelay                        = time.Millisecond * 100
+	defaultMinDelaySeconds               = int64(300)
+	defaultMaxDelaySeconds               = int64(600)
+	defaultBufferSizeLimitEvents         = int64(100)
+	defaultUploadSizeLimitBytes          = int64(32768) // 32KB limit
+	defaultBufferSizeLimitBytes          = int64(0)     // unlimited
+	defaultMaskDecisionPath              = "/system/log/mask"
+	defaultDropDecisionPath              = "/system/log/drop"
+	logRateLimitExDropCounterName        = "decision_logs_dropped_rate_limit_exceeded"
+	logNDBDropCounterName                = "decision_logs_nd_builtin_cache_dropped"
+	logBufferEventLimitExDropCounterName = "decision_logs_dropped_buffer_size_limit_events_exceeded"
+	logBufferSizeLimitExDropCounterName  = "decision_logs_dropped_buffer_size_limit_bytes_exceeded"
```

Comment:
while it is probably (also) helpful to have detailed counters for individual cases, from an (security) operational perspective, you are most likely interested how blind you actually are, as in how many decisions do not arrive at the control plane. 
Knowing how these graphs and alerts work, it would be helpful if we had just one counter that indicates that a decision was dropped independent of the reason. Otherwise operational graphs, metrics or alerts would easily miss the extra drops and would also need to be adjusted on configuration changes...

Reply:
Good point! Having only one counter makes sense to me. I renamed it to `decision_logs_dropped_buffer_size_limit_exceeded` dropping the `events` part. I don't think we can get rid of the existing type specific counter `decision_logs_dropped_buffer_size_limit_bytes_exceeded`, but I updated it so that the new generic counter gets updated in either scenario.